ipn/ipnlocal/local.go

```diff
@@ -7456,6 +7463,24 @@ func (b *LocalBackend) resetDialPlan() {
 	}
 }
 
+// getHardwareAddrs returns the hardware addresses for the machine. If the list
+// of hardware addresses is empty, it will return the previously known hardware
+// addresses. Both the current, and previously known hardware addresses might be
+// empty.
+func (b *LocalBackend) getHardwareAddrs() ([]string, error) {
+	addrs, err := posture.GetHardwareAddrs()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(addrs) == 0 {
```

Comment:
Might be good to log a message here if the cached set of addresses is not empty.